src/js/snapping.service.js

```diff
@@ -0,0 +1,84 @@
+import * as MathService from "./math.service";
+
+export function isLeftSnapping(firstPolygon, secondPolygon) {
+    const firstLine = MathService.initializeLine(secondPolygon.vertices[1], secondPolygon.vertices[2]);
+    const secondLine = MathService.initializeLine(firstPolygon.vertices[0], firstPolygon.vertices[3]);
+
+    const dx = firstPolygon.vertices[0].x - secondPolygon.vertices[1].x;
+
+    return MathService.isVerticalLinesSnapped(firstLine, secondLine) && Math.abs(dx) <= 10;
+}
+
+export function isRightSnapping(firstPolygon, secondPolygon) {
+    const firstLine = MathService.initializeLine(secondPolygon.vertices[0], secondPolygon.vertices[3]);
+    const secondLine = MathService.initializeLine(firstPolygon.vertices[1], firstPolygon.vertices[2]);
+
+    const dx = firstPolygon.vertices[1].x - secondPolygon.vertices[0].x;
+
+    return MathService.isVerticalLinesSnapped(firstLine, secondLine) && Math.abs(dx) <= 10;
+}
+
+export function isBottomSnapping(firstPolygon, secondPolygon) {
+    const firstLine = MathService.initializeLine(secondPolygon.vertices[0], secondPolygon.vertices[1]);
+    const secondLine = MathService.initializeLine(firstPolygon.vertices[3], firstPolygon.vertices[2]);
+
+    const dy = firstPolygon.vertices[2].y - secondPolygon.vertices[0].y;
+
+    return MathService.isHorizontalLinesSnapped(firstLine, secondLine) && Math.abs(dy) <= 10;
+}
+
+export function isTopSnapping(firstPolygon, secondPolygon) {
+    const firstLine = MathService.initializeLine(secondPolygon.vertices[3], secondPolygon.vertices[2]);
+    const secondLine = MathService.initializeLine(firstPolygon.vertices[0], firstPolygon.vertices[1]);
+
+    const dy = firstPolygon.vertices[0].y - secondPolygon.vertices[2].y;
+
+    return MathService.isHorizontalLinesSnapped(firstLine, secondLine) && Math.abs(dy) <= 10;
+}
+
+export function leftSnapping(firstPolygon, secondPolygon, mouse) {
```

Comment:
we need an action in this method call. smth like `snapToTheLeft`. the same in others